common/src/androidTest/kotlin/JSONArrayBugInstrumentedTest.kt

```diff
@@ -0,0 +1,33 @@
+import android.os.Build
+import com.revenuecat.purchases.common.networking.MapConverter
+import org.json.JSONObject
+import org.junit.Assert.assertEquals
+
+//@org.junit.runner.RunWith(androidx.test.ext.junit.runners.AndroidJUnit4::class)
+class JSONArrayBugInstrumentedTest {
+    @org.junit.Test
+    fun testReproduceIncorrectJSONArray() {
+        // these tests should only be run on Android 4, since that's the version that has the bug.
+        val osLevel = Build.VERSION.SDK_INT
+        if (!(osLevel >= Build.VERSION_CODES.ICE_CREAM_SANDWICH && osLevel <= Build.VERSION_CODES.KITKAT)) {
+            return
+        }
+
+        val inputMap = mapOf(
+            "key1" to listOf("value2", "value3"),
+        )
+        val incorrectJSONObject = JSONObject(inputMap)
+        assertEquals(
+            incorrectJSONObject
+                .toString(), ("{\"key1\":[\"value2, value3\"]}")
```

Comment:
this test on device is driving me a bit nuts, I might ask for someone to pair with me to verify whether I've lost my mind. 
Basically I get a mismatch between expected and actual... But the mismatch looks something like: 
expected `"{\"key1\":[\"value2, value3\"]}"`, got `"{\"key1\":[\"[value2, value3]\"]}"` (with another set of `[ ]`). 
The crazy part is that if I add those in, I get yet another level deeper, saying
expected `"{\"key1\":[\"value2, value3\"]}"`, got `"{\"key1\":[[\"[value2, value3]\"]]}"`. But I'm only changing the string literal here, not the input.

Reply:
@vegaro or @tonidero if you have any wisdom here or you can confirm that I've lot my mind, I'd be thankful either way

Reply:
Hmm I checked out your branch and indeed it was happening 😬 . I think this might be an issue with the `assertEquals` error message doing weird things (maybe an issue in junit 4.12?). When debugging, I noticed that we were checking against the wrong value and I was able to make the check pass by changing it to `assertEquals("{\"key1\":\"[value2, value3]\"}", incorrectJSONObject.toString())`. We could use `AssertJ` to do these assertions since it's what we use in our tests.